### ISSUE TITLE
False positive `inconsistent-return-statements` with `typing.assert_never`

### ISSUE THREAD

**Issue by jamesbraza** [pylint-dev/pylint#9921]
### Bug description

```python
from enum import StrEnum
from typing import assert_never


class SomeEnum(StrEnum):
    FOO = "foo"

    def do_something(self) -> str:
        if self == SomeEnum.FOO:
            return self.value
        assert_never(self)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module a
a.py:8:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
```

### Expected behavior

No `pylint` error due to `typing.assert_never` not needing a return.

It looks like https://github.com/pylint-dev/pylint/issues/7565 was related, but didn't fully fix.

### Pylint version

```shell
pylint 3.2.7
astroid 3.2.4
Python 3.12.5 (main, Aug 27 2024, 12:00:37) [Clang 15.0.0 (clang-1500.3.9.4)]
```

### OS / Environment

macOS Sonoma 14.5

**Comment by jamesbraza** [comment]
Cc @antoniogamizbadger since you made #9591

**Comment by jacobtylerwalls** [comment]
Thanks, can you retest against main?

**Comment by jacobtylerwalls** [comment]
#9591 has not been released yet.

**Comment by jamesbraza** [comment]
Shoot it passes 😆 

Sorry, I saw https://github.com/pylint-dev/pylint/releases/tag/v3.2.7 was out last week, and assumed #9591 was included by now. Sorry for the false positive bug report, closing this out
